shipment_management/public/js/get_rates.js

```diff
@@ -1,92 +1,109 @@
-        if (item.account_head == "Freight and Forwarding Charges - JA") {
-            frappe.throw("Shipment Charge has already been added")
-        }
-    })
-    frappe.call({
-        freeze: 1,
-        method: 'shipment_management.api.get_rates_for_doc',
-        args: {
-            doc: frm.doc
-        },
-        callback: function (response) {
-            let options_string = ""
-            let service_dict = {}
+	frm.doc.taxes.forEach((item) => {
+		if (item.account_head == "Freight and Forwarding Charges - JA") {
+			frappe.throw("Shipment Charge has already been added");
+		}
+	})
 
-            $.each(response.message, function (index, value) {
-                let option = value.label + " - $" + value.fee + "\n"
-                options_string += option
-                service_dict[option.slice(0, -1)] = {
-                    "label": value.label,
-                    "fee": value.fee
-                }
-            })
+	frappe.call({
+		freeze: 1,
+		method: 'shipment_management.shipengine.api.get_rates',
+		args: {
+			doc: frm.doc,
+			estimate: false
+		},
+		callback: function(response) {
+			let options = [];
+			let service_dict = [];
 
-            service_dict["PICK UP - $0"] = {"label" : "PICK UP", "fee" : 0}
-            options_string += "PICK UP - " + "$0" + "\n"
-            frappe.prompt({
-                    "label": "Service Types",
-                    "fieldtype": "Select",
-                    "fieldname": "service_types",
-                    "options": options_string.slice(0, -1),
-                    "reqd": 1
-                },
-                function (data) {
-                    let service_data = service_dict[data.service_types]
-                    if (service_data.fee != 0) {
-                        let row = frappe.model.add_child(frm.doc, "Sales Taxes and Charges", "taxes");
-                        row.charge_type = "Actual";
-                        row.account_head = "Freight and Forwarding Charges - JA";
-                        row.tax_amount = service_data.fee
-                        row.description = "Shipping(" + service_data.label + ")"
+			// form all the available shipping options
+			$.each(response.message, function(index, value) {
+				let estimated_arrival = isNaN(value.estimated_arrival) ? `(${value.estimated_arrival})` : `(${value.days} days)`;
+				let option = `${value.label} - $${value.fee} ${estimated_arrival}`;
 
-                    }
-                    frm.set_value("fedex_shipping_method", service_dict[data.service_types].label.replace(/ /g, "_"))
-                    refresh_field("taxes");
-                    frm.save()
-                },
-                "Select Service")
-        }
-    })
+				options.push(option);
+				service_dict.push({
+					"label": option,
+					"code": value.code,
+					"fee": value.fee,
+					"days": value.days,
+					"arrival": value.estimated_arrival
+				});
+			});
+
+			options.push("Pick Up - $0");
+			service_dict.push({
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'service_dict' is not in camel case.](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5409514)